app/views/courses/ManageLicenseModal.coffee

```diff
@@ -0,0 +1,261 @@
+      ))
+    return allPrepaids.map((p) ->
+      product = new Prepaid({
+        includedCourseIDs: p.productOptions.includedCourseIDs
+        type: 'course'
+      })
+      return {id: p.productId, name: product.typeDescription()}
+    )
+
+  enrolledUsers: ->
+    prepaid = @prepaidByGroup[@selectedPrepaidType]?.prepaid
+    return [] unless prepaid
+    @users.filter((user) ->
+      p = prepaid.numericalCourses()
+      s = p & user.prepaidNumericalCourses()
+      not (p ^ s)
+    )
+
+  unenrolledUsers: ->
+    prepaid = @prepaidByGroup[@selectedPrepaidType]?.prepaid
+    return [] unless prepaid
+    @users.filter((user) ->
+      p = prepaid.numericalCourses()
+      s = p & user.prepaidNumericalCourses()
+      p ^ s
+    )
+
+  allUsers: -> @users.toJSON()
+
+  areAllSelected: ->
+    return _.all(@unenrolledUsers(), (user) => @state.get('selectedUsers').get(user.id))
+
+  areAllSelectedRevoke: ->
+    return _.all(@allUsers(), (user) =>
+      @state.get('selectedUsers').get(user._id)
+    )
+
+  toggleSelectAllStudents: (e) ->
+    if @areAllSelected()
+      @unenrolledUsers().forEach (user, index) =>
+        if @state.get('selectedUsers').findWhere({ _id: user.id })
+          @$("[type='checkbox'][data-user-id='#{user.id}']").prop('checked', false)
+          @state.get('selectedUsers').remove(user.id)
+    else
+      @unenrolledUsers().forEach (user, index) =>
+        if not @state.get('selectedUsers').findWhere({ _id: user.id })
+          @$("[type='checkbox'][data-user-id='#{user.id}']").prop('checked', true)
+          @state.get('selectedUsers').add(user)
+
+  toggleSelectAllStudentsRevoke: (e) ->
+    if @areAllSelectedRevoke()
+      @users.forEach (user, index) =>
+        if @state.get('selectedUsers').findWhere({ _id: user.id })
+          @$("[type='checkbox'][data-user-id='#{user.id}']").prop('checked', false)
+          @state.get('selectedUsers').remove(user.id)
+    else
+      @users.forEach (user, index) =>
+        if not @state.get('selectedUsers').findWhere({ _id: user.id })
+          @$("[type='checkbox'][data-user-id='#{user.id}']").prop('checked', true)
+          @state.get('selectedUsers').add(user)
+
+  replaceStudentList: (e) ->
+    selectedClassroomID = $(e.currentTarget).val()
+    @classroom = @classrooms.get(selectedClassroomID)
+    if not @classroom
+      users = _.uniq _.flatten @classrooms.map (classroom) -> classroom.users.models
+      @users.reset(users)
+      @users.sort()
+    else
+      @users.reset(@classrooms.get(selectedClassroomID).users.models)
+    @render()
+    null
+
+  onSubmitForm: (e) ->
+    e.preventDefault()
+    @state.set error: null
+    usersToRedeem = @state.get('visibleSelectedUsers')
+    @redeemUsers(usersToRedeem)
+
+  updateVisibleSelectedUsers: ->
+    @state.set { visibleSelectedUsers: new Users(@state.get('selectedUsers').filter (u) => @users.get(u)) }
+
+  redeemUsers: (usersToRedeem) ->
+    if not usersToRedeem.size()
```

Comment:
Haven't checked, but is it `.size()` (here) or `.length`? (at the top of the template)?

Reply:
usersToRedeem is a backbone collection, it should support both length and size(). but i remember i use length get some error behaviros so size() is better maybe. (while backbone docs don't contains size() ..  strange..